### DISCUSSION TITLE
gopass include full file?

### DISCUSSION THREAD

**Discussion by davidolrik** [twpayne/chezmoi#1635] [Q&A] [answered]
Is there a way to include the full non-parsed contents of a gopass entry?

**Comment by twpayne** [comment]
I think the [`gopassRaw` template function](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md#gopassraw-gopass-name) is what you want.

**Reply by davidolrik** [reply]
It's close - but  `gopassRaw` is not actually raw.

When calling `show` gopass will [parse](https://github.com/gopasspw/gopass/blob/master/docs/commands/show.md#flags) anything that looks like yaml and allow you to do a secondary lookup like so:

```sh
$ gopass show entry key
```

This parsing normalises and reformats the yaml, which will break files where some of it look like yaml but isn't.

Here is an example of a ssh key

```diff
diff --git a/.ssh/id_rsa b/.ssh/id_rsa
index b00f644934b1c060711fe3cfa3c16410ffacd6a8..7ee99511a5375fa4a364d2472770988f732cd282 100600
--- a/.ssh/id_rsa
+++ b/.ssh/id_rsa
@@ -1,6 +1,6 @@
 -----BEGIN RSA PRIVATE KEY-----
-Proc-Type: 4,ENCRYPTED
-DEK-Info: AES-128-CBC,1748C5A19D5141CCE9D93B1A3246539A
+dek-info: AES-128-CBC,1748C5A19D5141CCE9D93B1A3246539A
+proc-type: 4,ENCRYPTED
```

I believe that `gopassRaw` should be fixed so it doesn't do any parsing, which will make gopass treat the whole secret as [plain](https://github.com/gopasspw/gopass/blob/master/docs/commands/show.md#parsing-and-secrets).

What do you think?

**Comment by twpayne** [comment, marked as answer]
Ah, thanks for the more detailed information. chezmoi's [current implementation of `gopassRaw` doesn't even include the `--password` flag](https://github.com/twpayne/chezmoi/blob/adfacfe902a1af1126476fafbe0645e23567e5b9/internal/cmd/gopasstemplatefuncs.go#L53) and so is likely broken, and I couldn't find any uses of the existing `gopassRaw` template function in any public dotfiles on GitHub or GitLab, so I think this is safe to change without fear of breaking backwards compatibility for anyone.

https://github.com/twpayne/chezmoi/pull/1636 contains the proposed fix. Does this look correct to you?

**Reply by davidolrik** [reply]
If you add `--password` it will be a breaking change, as that only returns the first line of the secret.
